### PR TITLE
Fix jack-o'-lantern orientation on placement. Closes GH-478

### PR DESCRIPTION
**Pull request by satoshinm**
Move the Pumpkin interface facing logic up to a BlockPumpkinBase class
shared by both BlockPumpkin and BlockJackOLantern. Placed
jack-o'-lanterns now face the player, just as pumpkins face the player.

### CODE REVIEW

File: src/main/java/net/glowstone/block/ItemTable.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ private void registerBuiltins() {
         reg(Material.TNT, new BlockTNT());
         reg(Material.DOUBLE_PLANT, new BlockDoublePlant());
         reg(Material.PUMPKIN, new BlockPumpkin());
-        reg(Material.JACK_O_LANTERN, new BlockDirectDrops(Material.JACK_O_LANTERN));
+        reg(Material.JACK_O_LANTERN, new BlockJackOLantern());
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/ItemTable.java]: Why create a new class instead of just using the constructor for BlockPumpkinBase?

**Review by mastercoms — CHANGES_REQUESTED** [review]

File: src/main/java/net/glowstone/block/blocktype/BlockPumpkinBase.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import net.glowstone.entity.GlowPlayer;
+import net.glowstone.inventory.ToolType;
+import org.bukkit.Material;
+import org.bukkit.block.BlockFace;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.material.MaterialData;
+import org.bukkit.material.Pumpkin;
+import org.bukkit.util.Vector;
+
+public abstract class BlockPumpkinBase extends BlockDirectDrops {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/blocktype/BlockPumpkinBase.java]: I don't think this class needs to be abstract.

**Review by mastercoms — CHANGES_REQUESTED** [review]

File: src/main/java/net/glowstone/block/ItemTable.java (hunk 1 above)
Comment (satoshinm) [reply src/main/java/net/glowstone/block/ItemTable.java]: I guess I could change it to `new BlockPumpkinBase(Material.JACK_O_LANTERN)`

File: src/main/java/net/glowstone/block/blocktype/BlockPumpkinBase.java (hunk 2 above)
Comment (satoshinm) [reply src/main/java/net/glowstone/block/blocktype/BlockPumpkinBase.java]: Removed abstract (was needed in an earlier iteration when I was changing the constructors but no longer necessary)

**Review by mastercoms — APPROVED** [review]

**Comment by mastercoms** [comment]
Looks good. Thanks for your contribution to Glowstone!

I'm a bit conflicted about merging this right now, since for thematic reasons it would be best to wait until Halloween.

**Pull request merged**